sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_testing_modules.md

```diff
@@ -0,0 +1,668 @@
+- **Regression detection**: Automatically catch when module behavior changes
+- **Comprehensive checking**: Validates file content, structure, and metadata
+
+## Your first simple test walkthrough with an example
+
+Let's walk through creating a basic test step-by-step using the `cat` module as an example:
+
+### Step 1: Examine the module
+
+First, let's look at a simple module (`modules/cat/main.nf`):
+
+```groovy
+process CAT {
+    input:
+    tuple val(meta), path(file_in)
+
+    output:
+    tuple val(meta), path("${prefix}"), emit: file_out
+    path "versions.yml"               , emit: versions
+
+    script:
+    prefix = "${meta.id}.txt"
+    """
+    cat ${file_in} > ${prefix}
+    """
+}
+```
+
+This process simply concatenates input files and outputs a single file.
+
+### Step 2: Write the test
+
+Create `modules/cat/tests/main.nf.test`:
+
+```groovy
+nextflow_process {
+    name "Test Process CAT"
+    script "../main.nf"
+    process "CAT"
+
+    test("simple file concatenation") {
+        when {
+            process {
+                """
+                input[0] = [
+                    [ id:'test' ], // meta map with sample identifier
+                    file('test_file.txt')  // input file
+                ]
+                """
+            }
+        }
+
+        then {
+            assert process.success               // Check process completed successfully
+            assert snapshot(process.out).match() // Validate outputs match expected snapshot
+        }
+    }
+}
+```
+
+**Understanding the test structure:**
+
+The nf-test file is organized into several key scopes that define what and how to test:
+
+- `nextflow_process { }` - **Test wrapper**: Declares this is a process-level test
+- `name` - **Test description**: Human-readable name for the test suite
+- `script` - **Module location**: Absolute/Relative path to the `main.nf` file containing the process
+- `process` - **Process name**: The specific process to test (must match the process name in `main.nf`)
+- `test("...")` - **Individual test case**: Each test block defines one scenario to test
+- `when { }` - **Test setup**: Where you define the input data for the process
+- `then { }` - **Assertions**: Where you verify the process behaved as expected
+
+**Inside the test components:**
+
+Now that we understand the overall structure, let's look at the specific elements within the test:
+
+- `[ id:'test' ]` - **meta map**: Contains sample metadata (ID, conditions, etc.)
+- `file('test_file.txt')` - **input file**: The file to process
+- `process.success` - **success check**: Ensures the process completed without errors
+- `snapshot(process.out).match()` - **output validation**: Compares all outputs to stored snapshot
+
+### Step 3: Run the test
+
+Once you've written the test, you can generate the initial snapshot by running the test.
+
+```bash
+cd path/to/modules
+nf-core modules test cat --profile docker (EXAMPLE)
+```
```

Comment:
Maybe not put `(EXAMPLE)` behind this as it looks like it's part of the command.